flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/pattern/Pattern.java

```diff
@@ -366,8 +366,9 @@ public Quantifier getQuantifier() {
 		checkIfNoNotPattern();
 		checkIfQuantifierApplied();
 		this.quantifier = Quantifier.times(quantifier.getConsumingStrategy());
-		if (from == 0) {
```

Comment:
Thanks for the suggestion, created PR: https://github.com/apache/flink/pull/4242